migrations/README.md

```diff
@@ -64,24 +107,49 @@ database with a given group of work, write a migration.
   - When in doubt, attach CSVs or screenshots from data clips generated from
   before the migration and after it was complete to the ticket.
 - Prefer having another engineer verify the data change, [[Four Eyes Principle]]
-- Use the `DataChange` utility to track changes.
 - Use SQL, not `ActiveRecord` models, in data migrations.
-- Depending on need data migrations fall into three use cases as follows.
 
+### Tracking Changes
+
+In order to know what was changed and when we should track any migrations run
+in our applications.
+
+- Use the `DataChange` functionality.
+- Scale data migrations based on the number of records being migrated.
+  - Keep in mind all environments and clients this migration could/will execute.
+  - With smaller data set changes
+    - Three steps in one `DataChange` call: `pre`, `post`, `change`.
+    - `pre` and `post` queries should include only records that are changing.
+    - `change` query include `returning` data.
+    -  All queries should return an identifier and the data that is migrating.
+  - With large data set changes
+    - The `pre` & `post` steps.
+      - Should be in two different `DataChange` calls.
+      - Could be full records or a summaries depending on the amount of data
+        being written to the `csv` files.
+    - Do not use the `change` `DataChange` step.
+    - Use the `find_in_batches` functionality to make the actual changes to
+      manage memory usage of the migration.
```

Comment:
I wonder about writing `puts` lines in each batch loop. Anecdotally running a long back fill without `puts` statements eventually got my session killed in gangway while adding `puts` kept it alive long enough to complete. Admittedly this was Gangway and I never learned a definitive reason why/how this would be the case, so take that with a grain of salt.

Reply:
I think this could be a good general practice that shouldn't hurt anything.